### DISCUSSION TITLE
Using a custom command for LSP config

### DISCUSSION THREAD

**Discussion by aneshodza** [Shopify/ruby-lsp#2760] [Q&A]
I have a very unique ruby setup, where `bundler exec` might not use the same version of ruby as I use locally. This still works for most things, except when the VSCode extension `ruby-lsp` tries to run its LSP:
```
Your Ruby version is 3.2.2, but your Gemfile specified 3.2.1 (Bundler::RubyVersionMismatch)
```
I have a custom command that would run it just fine:
```
BUNDLE_GEMFILE=~/global_gemfile bundle exec ruby-lsp
```
How can I make the plugin use my custom command to start ruby-lsp?

**Comment by andyw8** [comment]
@aneshodza Can you explain why you are not able to use the same version?

**Reply by aneshodza** [reply]
Using dockerised env. Cannot change that sadly and it's impossible to sync the versions up.

**Reply by d-lebed** [reply]
Potentially, this can be addressed here: https://github.com/Shopify/ruby-lsp/pull/2919

**Comment by vinistock** [comment]
This should already be possible by using our [custom bundle setting](https://github.com/Shopify/ruby-lsp/tree/main/vscode#using-a-custom-gemfile). Did you try using that already?

**Reply by aneshodza** [reply]
Yes I tried using a custom Gemfile. The Issue there was that it skipped my global one with a message telling me it found a local working gemfile and that it would skip my global one. I can retry tomorrow and keep you updated.
